### PR TITLE
Update readEvents() function

### PR DESCRIPTION
**Pull request by dadlex**
…g to avoid lack of Credentials as clientSDK uses default credentials while creating via URI.

Change magical number in list() method from 2 to 100. To allocate memory for 100 events. This will be changed to recycleLimit.
Change logic in read function as we now know that we may have more than one corrputed event

### CODE REVIEW

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 1]
```diff
@@ -67,37 +68,23 @@
 import io.pravega.client.stream.StreamConfiguration;
 import io.pravega.client.stream.TransactionalEventStreamWriter;
 import io.pravega.client.stream.TxnFailedException;
-import io.pravega.client.stream.impl.Controller;
-import io.pravega.client.stream.impl.ControllerImpl;
-import io.pravega.client.stream.impl.ControllerImplConfig;
+import io.pravega.client.stream.impl.*;
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: wildcard imports are not used in mongoose related projects, please consider to configure your ide

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 2]
```diff
@@ -67,37 +68,23 @@
...
-import java.util.ServiceLoader;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.Semaphore;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.TimeoutException;
+import java.util.*;
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: wildcard imports...

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 3]
```diff
@@ -830,28 +830,66 @@ void readEvents(final List<O> evtOps) {
...
+								}
+								Loggers.MSG.info("Scaling completed. Now amount of readers should be matched to {} segments",segments); //TODO: Change to debug
+							});
+
+							return evtReaderGroupName;
+						});
+				var evtReader_ = evtReaderPool.poll();
+
+				while (null == evtReader_) {
+					Thread.sleep(10000);
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: this is not looking like something could be

**Review by akurilov — CHANGES_REQUESTED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 1 above)
Comment (dadlex) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: I am sorry for that, I'm aware of this rule

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 3 above)
Comment (dadlex) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: while putting a thread to sleep is highly unobvious from a user's point of view, I wanted to use this approach to compensate for the user's mistakes

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 4]
```diff
@@ -862,42 +885,46 @@ void readEvents(final List<O> evtOps) {
...
-						completeOperation(evtOp, FAIL_TIMEOUT);
-					} else {
-						lastFailedStreamPos = streamPos;
-						Loggers.ERR.warn("{}: corrupted event @ position {}", stepId, streamPos);
-						completeOperation(evtOp, RESP_FAIL_CORRUPT);
-					}
-				} finally {
-					lastFailedStreamPosLock.unlock();
-				}
+		val evtRead=evtRead_;
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: fix formatting

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 5]
```diff
@@ -862,42 +885,46 @@ void readEvents(final List<O> evtOps) {
...
+		val evtData = evtRead.getEvent();
+		if (null == evtData) {
+			val streamPos = evtRead.getPosition();
+			if (((PositionImpl)streamPos).getOwnedSegments().isEmpty()) {
+				//means that reader doesn't own any segments, so it can't read anything
+				Loggers.MSG.debug("{}: empty reader. No EventSegmentReader assigned", stepId);
+                completeOperation(evtOp,PENDING);
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: fix formatting

File: src/test/robot/api/storage/pravega.robot [hunk 6]
```diff
@@ -122,7 +122,7 @@ Events Pipeline Test
     ...  --storage-net-node-addrs=${node_addr}
     ...  --storage-namespace=scope7 \
     ...  --storage-driver-limit-concurrency=10
-    ...  --storage-driver-threads=10
+    ...  --storage-driver-threads=1
```
Comment (akurilov) [inline src/test/robot/api/storage/pravega.robot]: what is the purpose of this change?

**Review by akurilov — CHANGES_REQUESTED** [review]

File: src/test/robot/api/storage/pravega.robot (hunk 6 above)
Comment (dadlex) [reply src/test/robot/api/storage/pravega.robot]: it's written in the commit name. I was checking if the problem with e2e latency test fail is correlated with the empty reader we get from a pool. This change proved that it is indeed correlated.
As it was said in the commit- it's only a check

File: README.md [hunk 7]
```diff
@@ -418,7 +442,7 @@ TEST=create_event_transactional_stream ./gradlew robotest
 cp -f build/libs/mongoose-storage-driver-pravega-*.jar ~/.mongoose/<MONGOOSE_BASE_VERSION>/ext/
 ```
 Note that the Pravega storage driver depends on the 
-[Coop Storage Driver](http://repo.maven.apache.org/maven2/com/github/emc-mongoose/mongoose-storage-driver-coop/) 
+[Preempt Storage Driver](http://repo.maven.apache.org/maven2/com/github/emc-mongoose/mongoose-storage-driver-preempt/)
```
Comment (akurilov) [inline README.md:445]: Please consider such change also in the "Deployment -> Basic" section.
Then you can skip ci (if you update the documentation only)

Comment (dadlex) [reply README.md:445]: This change is on master already. I just noticed that in this section it didn't get updated.

**Review by akurilov — APPROVED** [review]
